cpp/src/join/conditional_join_kernels.cuh

```diff
@@ -242,6 +242,7 @@ __global__ void conditional_join(table_device_view left_table,
                         join_shared_r[warp_id]);
     }
 
+    __syncwarp(activemask);
     // final flush of output cache
     if (current_idx_shared[warp_id] > 0) {
       flush_output_cache<num_warps, output_cache_size>(activemask,
```

Comment:
I wonder if the actually problem again is the fact that the `activemask` being passed to `flush_output_cache` is wrong since we are in a `if` conditonal. I think we need a ballot_sync on `current_idx_shared[warp_id] > 0` to compute the correct submask

Reply:
So, I tried this but verified the `syncwarp` is still required here. I will push a proper fix for the mask handling.